pythainlp/util/normalize.py

```diff
@@ -79,14 +78,20 @@ def remove_tonemarks(text: str) -> str:
         delete_tone('สองพันหนึ่งร้อยสี่สิบเจ็ดล้านสี่แสนแปดหมื่นสามพันหกร้อยสี่สิบเจ็ด')
         # output: สองพันหนึงรอยสีสิบเจ็ดลานสีแสนแปดหมืนสามพันหกรอยสีสิบเจ็ด
     """
-    return _RE_TONEMARKS.sub("", text)
+    for ch in tonemarks:
+        while ch in text:
+            text = text.replace(ch, "")
+    return text
 
 
 def remove_zw(text: str) -> str:
     """
     Remove zero-width characters.
     """
-    return _RE_REMOVE_ZERO_WIDTHS.sub("", text)
+    for ch in _ZERO_WIDTH_CHARS:
+        while ch in text:
+            text = text.replace(ch, "")
+    return text
```

Comment:
just out of my curiosity, what is the disadvantage of the previous regex solution?

Reply:
Tested with several strings, with the code below, the str.replace in while loop one is around twice faster.

```python
_ZERO_WIDTH_CHARS = "\u200b\u200c"
_RE_REMOVE_ZERO_WIDTHS = re.compile(f"[{_ZERO_WIDTH_CHARS}]+")
def rmz_re(text: str) -> str:
    return _RE_REMOVE_ZERO_WIDTHS.sub("", text)

def rmz_replace(text: str) -> str:
    for ch in _ZERO_WIDTH_CHARS:
        while ch in text:
            text = text.replace(ch, "")
    return text
```